kafka_consumer/datadog_checks/kafka_consumer/client.py

```diff
@@ -150,99 +153,76 @@ def get_consumer_offsets(self):
 
     def _get_consumer_groups(self):
-        # Get all consumer groups
+        # Get all consumer groups to monitor
         consumer_groups = []
-        consumer_groups_future = self.kafka_client.list_consumer_groups()
-        self.log.debug('MONITOR UNLISTED CG FUTURES: %s', consumer_groups_future)
-        try:
-            list_consumer_groups_result = consumer_groups_future.result()
-            self.log.debug('MONITOR UNLISTED FUTURES RESULT: %s', list_consumer_groups_result)
+        if self.config._monitor_unlisted_consumer_groups or self.config._consumer_groups_compiled_regex:
+            consumer_groups_future = self.kafka_client.list_consumer_groups()
+            try:
+                list_consumer_groups_result = consumer_groups_future.result()
+
+                consumer_groups.extend(
+                    valid_consumer_group.group_id for valid_consumer_group in list_consumer_groups_result.valid
+                )
+            except Exception as e:
+                self.log.error("Failed to collect consumer groups: %s", e)
+            return consumer_groups
+        else:
+            return self.config._consumer_groups
 
-            consumer_groups.extend(
-                valid_consumer_group.group_id for valid_consumer_group in list_consumer_groups_result.valid
-            )
-        except Exception as e:
-            self.log.error("Failed to collect consumer groups: %s", e)
-        return consumer_groups
+    def _list_consumer_group_offsets(self, cg_tp):
+        return self.kafka_client.list_consumer_group_offsets([cg_tp])
 
     def _get_consumer_offset_futures(self, consumer_groups):
-        topic_metadata = self.kafka_client.list_topics(timeout=self.config._request_timeout).topics
-        topics = {
-            topic: list(topic_metadata[topic].partitions.keys())
-            for topic in topic_metadata
-            if topic not in KAFKA_INTERNAL_TOPICS
-        }
+        futures = []
 
-        for consumer_group in consumer_groups:
-            self.log.debug('CONSUMER GROUP: %s', consumer_group)
+        # If either monitoring all consumer groups or regex, return all consumer group offsets (can filter later)
+        if self.config._monitor_unlisted_consumer_groups or self.config._consumer_groups_compiled_regex:
+            for consumer_group in consumer_groups:
+                futures.append(
+                    self._list_consumer_group_offsets(ConsumerGroupTopicPartitions(consumer_group))[consumer_group]
+                )
+            return futures
 
-            for topic_partition in self._get_topic_partitions(topics, consumer_group):
-                yield self.kafka_client.list_consumer_group_offsets(
-                    [ConsumerGroupTopicPartitions(consumer_group, [topic_partition])]
-                )[consumer_group]
+        for consumer_group in consumer_groups:
+            # If topics are specified
+            if topics := consumer_groups[consumer_group]:
```

Comment:
And then dedent below

```suggestion
            topics = consumer_groups.get(consumer_group):
            if not topics:
                continue
```